### ISSUE TITLE
find d 2012 and p england

### ISSUE THREAD

**Issue by kaplun** [inveniosoftware/invenio-query-parser#20]
So: I really think we should no longer work on INSPIRE-related search engine functionalities for Invenio 1.x. So I am moving invenio-software/invenio#718 to this repository.

`find d 2012 and p england` is currently interpreted by query parser as:
```python
AndOp(KeywordOp(Keyword('year'), Value('2012')), KeywordOp(Keyword('year'), Value('p england')))
```

Is this the correct interpretation? @tsgit, @tiborsimko?

**Comment by tiborsimko** [comment]
I don't know whether `p` stands for some logical field in SPIRES; Annette or Heath will be able to tell.  
If yes, we should add it, and fix the above expansion not to use `year` after `and`.  If not, the above expansion may be acceptable; though it is not user friendly, since it would lead to no hits found.  

(Basically, the question is whether `d` should be taken over after `and`, which I think was the case in SPIRES,  due to no global index, but which we may want to avoid now and look for after-and part in the global index perhaps...)

**Comment by tiborsimko** [comment]
See also similar issues:
- https://github.com/inveniosoftware/invenio-query-parser/issues/16#issuecomment-86943861
- https://github.com/inveniosoftware/invenio-query-parser/issues/6#issue-47382000

**Comment by kaplun** [comment]
> I don't know whether p stands for some logical field in SPIRES; Annette or Heath will be able to tell.

OK, I have checked the code base in Invenio legacy and in the documentation, at https://inspirehep.net/info/hep/search-tips#basic

There are no traces of `p` element.

**Comment by tiborsimko** [comment]
Yeah, but not every field has been listed in the docs or present in `search_engine_query_parser.py` from the start, so it would be good to double check with Annette/Kirsten/Heath, just to be sure...

**Comment by kaplun** [comment]
OK checking the original RT ticket it looks like the intended query wanted to find all conferences in England in 2012.

So the trick is simply to support an alias between "p" and whatever is the definition of a place.

**Comment by kaplun** [comment]
@jacquerie this is going to be part of the search functionality work we are planning.
